services/director-v2/tests/unit/test_modules_dynamic_sidecar_scheduler_task.py

```diff
@@ -96,8 +99,42 @@ def increment(self) -> None:
         self.count += 1
 
 
+@pytest.fixture(params=[True, False])
+def error_raised_by_saving_state(request: FixtureRequest) -> bool:
+    return request.param  # type: ignore
+
+
+@dataclass
+class UseCase:
+    can_save: bool
+    were_containers_created: bool
+    outcome_service_removed: bool
+
+
+@pytest.fixture(
+    params=[
+        UseCase(
+            can_save=False, were_containers_created=False, outcome_service_removed=True
+        ),
+        UseCase(
+            can_save=False, were_containers_created=True, outcome_service_removed=True
+        ),
+        UseCase(
+            can_save=True, were_containers_created=False, outcome_service_removed=True
```

Comment:
ok and my question in that case: can_save is True, will it actually save the state even though the containers were not created? If yes why?

Reply:
It will not save the state here. Since the save stata only triggers if:
```python
# only try to save the status if :
# - it is requested to save the state
# - the dynamic-sidecar has finished booting correctly
if (
    scheduler_data.dynamic_sidecar.service_removal_state.can_save
    and scheduler_data.dynamic_sidecar.were_containers_created
):
```